client/evilwallet/wallets.go

```diff
@@ -80,20 +93,45 @@ func (w *Wallets) GetUnspentOutput(wallet *Wallet) *Output {
 	return wallet.GetUnspentOutput()
 }
 
-// FreshWallet gets a fresh faucet wallet.
-func (w *Wallets) FreshWallet() *Wallet {
-	if len(w.faucetWallets) > 0 {
-		return w.wallets[w.faucetWallets[0]]
+// IsFaucetWalletAvailable checks if there is any faucet wallet left.
+func (w *Wallets) IsFaucetWalletAvailable() bool {
+	return len(w.faucetWallets) > 0
+}
+
+// FreshWallet returns the first non-empty wallet from the faucetWallets queue. If current wallet is empty,
+// it is removed and the next enqueued one is returned.
+func (w *Wallets) FreshWallet() (*Wallet, error) {
+	wallet, err := w.GetNextWallet(fresh)
+	if err != nil {
+		w.removeWallet(fresh)
+		wallet, err = w.GetNextWallet(fresh)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
I think these logics should be in GetNextWallet, it should iterate the whole wallet list, remove the empty one and return one if it's usable.

Reply:
Why do we want to iterate over this list int the first place? As this list is basically a queue, new wallets are added at the end, currently used wallet is at index 0, if is used up we get next from queue. We never use wallets at index different than 0.

I see that I did not add switch for wallet type
Also I will add a  rule for `SetWalletReady` to not add empty wallets just in case.

Reply:
I mean something like this in `GetWallet`:
```go
for _, wallet := range w.wallets {
    if wallet.IsEmpty() {
	w.removeWallet(fresh)
    }
    return w, nil
}

return nil, errors.New("wallet type not supported for ordered usage, use GetWallet by ID instead")
```